kythe/cxx/verifier/souffle_interpreter.h

```diff
@@ -17,8 +17,24 @@
 #ifndef KYTHE_CXX_VERIFIER_SOUFFLE_INTERPRETER_H_
 #define KYTHE_CXX_VERIFIER_SOUFFLE_INTERPRETER_H_
 
+#include "kythe/cxx/verifier/assertions.h"
+
 namespace kythe::verifier {
-bool RunSouffle();
+class Verifier;
+/// \brief Runs the Souffle interpreter on parsed facts and goal groups.
+/// \param symbol_table the symbol table for expanding idents
+/// \param goal_groups the goal groups to solve
+/// \param database the facts to solve against
+/// \param verifier the verifier to pass to the inspection
+/// \param inspect the inspection to use
+/// \param highest_goal_reached will be set to the highest goal reached
+/// \param highest_group_reached will be set to the highest group reached
+/// \return true if all goals could be solved
+bool RunSouffle(
+    const SymbolTable& symbol_table, const std::vector<GoalGroup>& goal_groups,
+    const Database& database, Verifier* verifier,
+    std::function<bool(Verifier*, const AssertionParser::Inspection&)> inspect,
+    size_t& highest_goal_reached, size_t& highest_group_reached);
```

Comment:
Prefer a return value for these, something like:

struct SouffleResult {
  bool success;
  size_t highest_goal_reached;
  size_t highest_group_reached;
};